### ISSUE TITLE
openstack_networking_secgroup_v2 import does not honour "delete_default_rules" flag

### ISSUE THREAD

**Issue by domdom82** [terraform-providers/terraform-provider-openstack#549]
### Terraform Version
Terraform v0.11.11

### Affected Resource(s)
- openstack_networking_secgroup_v2

### Terraform Configuration Files
```
resource "openstack_networking_secgroup_v2" "emptyrule" {
  name        = "emptyrule"
  description = "this should be empty"
  delete_default_rules = true
}
```

### Expected Behavior
`terraform import "openstack_networking_secgroup_v2.emptyrule" <uuid>`
should import an empty rule with the attribute "delete_default_rules" set to "true".

`terraform plan` should report no changes after the import.

### Actual Behavior
The attribute "delete_default_rules" is missing.

`terraform plan` reports these changes:

```
Terraform will perform the following actions:

-/+ openstack_networking_secgroup_v2.emptyrule (new resource required)
      id:                   "<uuid>" => <computed> (forces new resource)
      delete_default_rules: "" => "true" (forces new resource)
      description:          "this should be empty" => "this should be empty"
      name:                 "emptyrule" => "emptyrule"
      region:               "<region>" => <computed>
      tenant_id:            "<uuid>" => <computed>


Plan: 1 to add, 0 to change, 1 to destroy.
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform import "openstack_networking_secgroup_v2.emptyrule"`
1. `terraform plan`

**Comment by domdom82** [comment]
Verified this bug exists with latest versions of tf and the provider pulled from master.

I'd like to fix this myself, can someone point me to the right spot in the code where the actual import happens? Could not find it in the provider and the interactions between tf and providers are kind of opaque to me atm.

**Comment by domdom82** [comment]
hi @jtopjian thanks for the quick reply. I get your point, though it breaks my use-case.
I am writing a manifest that creates security groups (duh) which can be created anew or be imported from a legacy tool that used the OS CLI to create them.

The migration into TF should be as such that we list OS resources via the CLI, then map them to TF resources and import them. Finally to verify that everything went well, we do a `terraform plan` to make sure the imported resources look just like they would if we had deployed them via TF natively.

Usually, this works fine because the resource attributes can be deduced from the actual resources on the cloud. In this case however, `delete_default_rules` is a "TF-only" attribute that disappears once the resource has been deployed. To make my migration work now, I have to fumble this attribute manually into the tfstate file so that the `terraform plan` does not show changes (which results in rc != 0, which fails my migration script).

You could argue that I should just accept the change and do a `terraform apply` to heal the state again, but this would mean not just a change of the existing resource, but destruction and recreation, which in the case of security groups can lead to a downtime.

tldr; I can't remove the `delete_default_rules` attribute because I need it for the "from-scratch" use-case. Not sure if we can make a change to this attribute not force a resource recreation either. Could `terraform import` just ignore it or write it to state silently? Since the attribute makes no sense for importing stuff, we could simply fix the state so that it does not show changes?

**Comment by jtopjian** [comment]
It might be possible to do this. You might have some luck looking at other Terraform providers (such as the AWS provider) to see if there's prior art for this kind of behavior.

**Comment by jtopjian** [comment]
Thinking out loud: setting aside if resolving this is possible to do, there's a good chance you might run into other quirks like this. Depending on how you're setting up this system of yours, it might be beneficial to have some kind of variable / flag that a user would set depending on if they are importing existing resources or starting from scratch.

You could then use [conditionals](https://www.terraform.io/docs/configuration/interpolation.html#conditionals) to set values like this appropriately.

While I'm not dismissing this `delete_default_rules` issue (though, to be clear: I am not sure at all if this is something fixable), this kind of flag would unblock you from proceeding for the time being.

**Comment by domdom82** [comment]
@jtopjian worked around the problem using conditionals. Thanks!
